frontend/coffee/profile.coffee

```diff
@@ -31,3 +35,65 @@ window.upload_bg = (files, box) ->
     formdata = new FormData()
     formdata.append('image', file)
     xhr.send(formdata)
+
+
+# Handling of the change-password-dialog
+document.getElementById('change-password-submit').addEventListener('click', (e) ->
+    e.preventDefault()
+
+    # Disable the buttons until we get an response
+    buttons = document.getElementsByClassName('btn-pw-change')
+    for button in buttons
+        button.setAttribute('disabled', '')
+
+    old_password = get('old-password')
+    new_password = get('new-password1')
+    new_password_confirm = get('new-password2')
+
+    xhr = new XMLHttpRequest()
+    xhr.open('POST', "/api/user/#{window.username}/change-password")
+    # Triggered after we get an response from the server.
+    # It's in the form {'error': bool, 'reason': string)
+    xhr.onload = () ->
+        result = JSON.parse(this.responseText)
+        error_message_display = document.getElementById('error-message')
+
+        if result.error == true
+            error_message_display.innerHTML = result.reason
+            error_message_display.classList.add('text-danger')
+            error_message_display.classList.remove('hidden')
+            # Re-enable the buttons.
+            for button in buttons
+                button.removeAttribute('disabled')
+        else
+            error_message_display.innerHTML = 'Password changed successfully.'
+            error_message_display.classList.remove('text-danger')
+            error_message_display.classList.add('text-success')
+            error_message_display.classList.remove('hidden')
+            # .modal('hide') doesn't work. Let's reload the page instead.
+            # Delay it a bit to give the user a chance to read the response message.
+            setTimeout((() -> window.location = window.location), 1000)
+
+    form = new FormData()
+    form.append('old-password', old_password)
+    form.append('new-password', new_password)
+    form.append('new-password-confirm', new_password_confirm)
+    xhr.send(form)
+)
+
+
+resetPasswordModalDialog = () ->
+    $('#password-form').trigger('reset')
+    error_message_display = document.getElementById('error-message')
+    error_message_display.innerHTML = ''
+    error_message_display.classList.remove('text-danger')
+    error_message_display.classList.remove('text-success')
+    error_message_display.classList.add('hidden')
+
+    buttons = document.getElementsByClassName('btn-pw-change')
+    for button in buttons
+        button.removeAttribute('disabled')
+
+$('#change-password').on('hidden.bs.modal', () ->
+    resetPasswordModalDialog()
+)
```

Comment:
```suggestion
$('#change-password').on('hidden.bs.modal', resetPasswordModalDialog)
```

I think this will work without the extra lambda.

Reply:
Probably true. I'll try it out to be sure.